packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -358,54 +360,96 @@ - (NSNumber *)getDesiredImageQuality:(NSNumber *)imageQuality {
 - (void)picker:(PHPickerViewController *)picker
     didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14)) {
   [picker dismissViewControllerAnimated:YES completion:nil];
-
+  if (results.count == 0) {
+    self.result(nil);
+    self.result = nil;
+    _arguments = nil;
+    return;
+  }
   NSNumber *maxWidth = [_arguments objectForKey:@"maxWidth"];
   NSNumber *maxHeight = [_arguments objectForKey:@"maxHeight"];
   NSNumber *imageQuality = [_arguments objectForKey:@"imageQuality"];
   NSNumber *desiredImageQuality = [self getDesiredImageQuality:imageQuality];
+  NSMutableArray *pathList = [[NSMutableArray alloc] initWithCapacity:results.count];
+  for (int i = 0; i < results.count; [pathList addObject:[NSNull null]], i++)
+    ;
```

Comment:
@mvanbeusekom and @cyanglaz Do you know a better way to create an array in which you can already add objects in random order. If we don't add this for-loop the app crashes when image 1 is completed before image 0 and we want to add it like this: `pathList[i] = savedPath;`

Reply:
@renefloor is this what you mean:

```objc
    NSString * strings [3]; // Declare an array of three elements
    strings[2] = @"World"; // Assign the third element
    strings[0] = @"Hello"; // Assign the first element
    strings[1] = @" "; // Assign the second element
    int i = 0;
    while(i<3) {
        NSLog(strings[i]);
        i++;
    }
```

Reply:
Didn't think `NSString * strings [3];` would work, was searching for something like `[[NSMutableArray alloc] initWithCapacity:results.count];`

Reply:
Doing this with an array was also not really easily possible as we need it in the block and

> __block attribute not allowed on declaration with a variably modified type

So now we moved the creation of the array to a documented function.